lib/std/zig/system.zig

```diff
@@ -224,7 +224,31 @@ pub const NativeTargetInfo = struct {
                     // TODO Detect native operating system version.
                 },
                 .macosx => {
-                    // TODO Detect native operating system version.
+                    var product_version: [32]u8 = undefined;
+                    var size: usize = @sizeOf(@TypeOf(product_version));
```

Comment:
Better to use product_version.len here. If the type was changed to e.g. a slice then it would still be correct instead of a bug.